addons/binding/org.openhab.binding.tivo/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,152 @@
+control of channel changes, user interface navigation and allow the client to send simulated remote control button presses to the Digital Video Recorder.<![CDATA[ <br /><br /> ]]>See https://www.tivo.com/assets/images/abouttivo/resources/downloads/brochures/TiVo_TCP_Network_Remote_Control_Protocol.pdf for supported commands / codes for each channel.</description>
+
+        <channels>
+            <channel id="tivoChannelForce" typeId="tivoChannelForce"/>
+            <channel id="tivoChannelSet" typeId="tivoChannelSet"/>
+            <channel id="tivoTeleport" typeId="tivoTeleport"/>
+            <channel id="tivoIRCommand" typeId="tivoIRCommand"/>
+            <channel id="tivoKBDCommand" typeId="tivoKBDCommand"/>
+            <channel id="tivoStatus" typeId="tivoStatus"/>
+            <channel id="tivoCommand" typeId="tivoCommand"/>
+        </channels>
+        
+        <config-description>
+        	<parameter name="deviceName" type="text" required="true">
+				<default>My Tivo</default>
+				<required>true</required>
+				<label>Device Name</label>
+				<description>A friendly name to refer to this device.<![CDATA[ <b> ]]>Default: Device name specified on the TiVo.<![CDATA[ </b> ]]></description>
+			</parameter>
+        	<parameter name="address" type="text" required="true">
+            	<default></default>
+            	<required>true</required>
+            	<label>Address</label>
+            	<description>The IP address or host name of your TiVo box</description>
+        	</parameter>
+        	<parameter name="tcpPort" type="integer" max="65535" min="1">
+				<default>31339</default>
+				<required>true</required>
+				<label>TCP Port</label>
+				<context>network-address</context>
+				<description>The TCP port number used to connect to the TiVo.<![CDATA[ <b> ]]>Default: 31339<![CDATA[ </b> ]]></description>
+			</parameter>
+			<parameter name="numRetry" type="integer" max="20" min="0">
+				<default>5</default>
+				<required>true</required>
+				<label>Connection Retries</label>
+				<description>The number of times to attempt reconnection to the TiVo box, if there is a connection failure.<![CDATA[ <b> ]]>Default: 5<![CDATA[ </b> ]]></description>
+			</parameter>
+			<parameter name="keepConActive" type="boolean">
+				<default>true</default>
+				<required>true</required>
+				<label>Keep Connection Open</label>
+				<description>Keep connection to the TiVo open.  Recommended for monitoring the TiVo for changes in TV channels.  Disable if other applications that use the Remote Control Protocol port will also be used e.g. mobile remote control applications.<![CDATA[ <b> ]]>Default: True (Enabled)<![CDATA[ </b> ]]></description>
+			</parameter>
+			<parameter name="pollForChanges" type="boolean">
+				<default>true</default>
```

Comment:
Ooops, sorry confused myself here.  "The TiVo device state when you connect is non-existent so I have come up with a workaround for this (but it is a bit slower than I you would like)" is not true.

Each time you connect the TiVo responds with the current channel status (online) or nothing (standby).  I work out the standby status of the box by sending a 'garbage' command to the TiVo when no current channel is received on connection, if I get `INVALID_COMMAND` then the system is in standby.  If I get nothing then OFFLINE.

So what I think you were getting at is when you have keepConnActive (recommended) does the TiVo update the status when the remote is used instead of OH.  The answer is yes.  

So I could get rid of the "10 sec" polling job in this setup, if I could work out a way to detect when new data was in the `BufferedReader` and trigger a new status update based on that event.

Now I did look into this, but I could not work out a way to do it.  I'm sure this must be possible, but my google foo failed me... so I ended up just polling every 10 secs to 'catch' these.  This is the main reason why `TivoStaytusProvider#statusGetRetry()` always only looks for the last status update line.

If there is a way to do this, can you point me to the right techniques or a binding that does this, so I can work it out?

A